packages/analytics-js/public/index.html

```diff
@@ -35,15 +36,117 @@
         })(method);
       }
 
+      // New addition to load script starts
+      var sdkBuildType = 'legacy';
+
+      try {
+        // Feature detection of dynamic imports
+        new Function('return import("")');
+        sdkBuildType = 'modern';
+      } catch (e) {
+      }
+
+      var rudderCdnBaseUrlHost = 'https://cdn.rudderlabs.com/beta/v3/'
+      var rudderAnalyticsBaseUrl = rudderCdnBaseUrlHost + sdkBuildType;
+      var rudderAnalyticsMount = function() {
+        var rudderAnalyticsScript = document.createElement('script');
+        rudderAnalyticsScript.src = rudderAnalyticsBaseUrl + '/rudder-analytics.min.js';
+        rudderAnalyticsScript.async = true;
+        rudderAnalyticsScript.defer = true;
+        // document.head.appendChild(rudderAnalyticsScript);
+      };
+
+      // Feature detection of promise
+      if(typeof Promise === 'undefined') {
+        var rudderAnalyticsPromisesScript = document.createElement('script');
+        rudderAnalyticsPromisesScript.src = 'https://polyfill.io/v3/polyfill.min.js?features=Promise&callback=rudderAnalyticsMount';
+        rudderAnalyticsPromisesScript.async = true;
+        rudderAnalyticsPromisesScript.defer = true;
+        document.head.appendChild(rudderAnalyticsPromisesScript);
+      } else {
+        rudderAnalyticsMount();
+      }
```

Comment:
Is it possible to bundle the Promise polyfill alone to avoid this?

Reply:
it bloats up the size to 48kb.

at the moment it will do for the beta, in the future I want to do this into the js code bore new RudderAnalytics instance is created.

Unfortunattelly the Promise polyfill is required as soon as out js code is getting evaluated at the moment and cannot change this.

Reply:
Sounds good. Added to our list for considering this later.